### PR TITLE
Refactor ExtendedAPIPlugin in ES6

### PR DESCRIPTION
**Pull request by nicksp**
**What kind of change does this PR introduce?**

Refactor

**Did you add tests for your changes?**

N/A - no new functionality.

**If relevant, link to documentation update:**

N/A

**Summary**

Refactor `ExtendedAPIPlugin.js` to ES 2015.

**Does this PR introduce a breaking change?**

No

**Other information**

### CODE REVIEW

File: lib/ExtendedAPIPlugin.js [hunk 1]
```diff
@@ -3,40 +3,40 @@
...
-		params.normalModuleFactory.plugin("parser", function(parser, parserOptions) {
-			Object.keys(REPLACEMENTS).forEach(function(key) {
-				parser.plugin("expression " + key, ParserHelpers.toConstantDependency(REPLACEMENTS[key]));
-				parser.plugin("evaluate typeof " + key, ParserHelpers.evaluateToString(REPLACEMENT_TYPES[key]));
+class ExtendedAPIPlugin {
+	apply(compiler) {
+		compiler.plugin("compilation", (compilation, params) => {
+			compilation.dependencyFactories.set(ConstDependency, new NullFactory());
+			compilation.dependencyTemplates.set(ConstDependency, new ConstDependency.Template());
+			compilation.mainTemplate.plugin("require-extensions", function(source, chunk, hash) {
```
Comment (lior-a) [inline lib/ExtendedAPIPlugin.js]: You should be able and use arrow function here

File: lib/ExtendedAPIPlugin.js [hunk 2]
```diff
@@ -3,40 +3,40 @@
...
+				var buf = [source];
```
Comment (lior-a) [inline lib/ExtendedAPIPlugin.js]: unless you need to scope var here I would use let.

File: lib/ExtendedAPIPlugin.js [hunk 3]
```diff
@@ -3,40 +3,40 @@
...
+				buf.push("");
+				buf.push("// __webpack_hash__");
+				buf.push(this.requireFn + ".h = " + JSON.stringify(hash) + ";");
```
Comment (lior-a) [inline lib/ExtendedAPIPlugin.js]: how about template literals ?

**Comment by sendilkumarn** [comment]
This was already done in #4361 @nicksp

File: lib/ExtendedAPIPlugin.js (hunk 3 above)
Comment (nicksp) [inline lib/ExtendedAPIPlugin.js]: Done

File: lib/ExtendedAPIPlugin.js (hunk 2 above)
Comment (nicksp) [inline lib/ExtendedAPIPlugin.js]: Done

File: lib/ExtendedAPIPlugin.js (hunk 1 above)
Comment (nicksp) [inline lib/ExtendedAPIPlugin.js]: I guess an `arrow fn` isn't applicable here, and we have to apply a function expression, which allows to change `this` depending on the target element.

**Comment by nicksp** [comment]
It makes much more sense to have one thing per commit rather than the whole bunch of stuff. It's more profitable to the app as allows to rollback one thing at a time if PR didn't work well os whatever...

**Comment by sendilkumarn** [comment]
Isn't it one thing per commit 😕

**Comment by sokra** [comment]
Thanks

**Pull request merged**